libs/menureq.py

```diff
@@ -620,6 +620,15 @@ def get_top_pal(pal, colorlist, num_colors, halfbright):
 
         return newpal
 
+    def get_max_color_index():
+        max_color_index = 0
+        for y in range(config.pixel_canvas.get_height()):
+            for x in range(config.pixel_canvas.get_width()):
+                color_index = config.pixel_canvas.get_at_mapped((x,y))
+                if max_color_index < color_index:
+                    max_color_index = color_index
```

Comment:
I'd replace this with a pygame.surfarray.pixels2d() call and a max() function to let numpy do the work instead of slow loops in Python and slow calls to get_at_mapped().